lib/common/mobile/ios/ios-device-base.ts

```diff
@@ -24,8 +24,8 @@ export abstract class IOSDeviceBase implements Mobile.IiOSDevice {
 			this.cachedSockets[appId] = await this.getDebugSocketCore(appId, projectName);
 
 			if (this.cachedSockets[appId]) {
-				this.cachedSockets[appId].on("close", () => {
-					this.destroyDebugSocket(appId);
+				this.cachedSockets[appId].on("close", async () => {
+					await this.destroyDebugSocket(appId);
```

Comment:
This callback will not be awaited and the lock might be released too early. Maybe just resolve a promise outside of the event handler scope after the callback is called and  the result of `this.destroyDebugSocket(appId)` is resolved.

Reply:
It will be an infinite wait if we wait for the close event in the create socket method. We are attaching here in order to clean up the cache when the socket is closed.